### PR TITLE
Task/m fc datasource all imaged clusters

### PR DESCRIPTION
**Pull request by abhimutant**
datasource all imaged clusters

### CODE REVIEW

File: nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+						},
+						"imaged_node_uuid_list": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"common_network_settings": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Optional: true,
```
Comment (siddharth-kulshrestha) [inline nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go:64]: Do we really need to mark these fields as optional?

Comment (siddharth-kulshrestha) [reply nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go:64]: Please review all the fields in this file.

File: nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+	if offset, offok := d.GetOk("offset"); offok {
+		req.Offset = utils.IntPtr(offset.(int))
+	}
+
+	if filter, fok := d.GetOk("filters"); fok {
+		filt := &fc.ImagedClustersListFilter{}
+		filter := filter.([]interface{})[0].(map[string]interface{})
+		filt.Archived = utils.BoolPtr(filter["archived"].(bool))
+		req.Filters = filt
+	}
```
Comment (siddharth-kulshrestha) [inline nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go:467]: According to this implementation length, offset and filters should be optional

**Review by siddharth-kulshrestha — APPROVED** [review]
Minor comments which can be fixed in the next PR

File: nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go (hunk 1 above)
Comment (abhimutant) [reply nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go:64]: Sure will review and remove optional fields.

File: nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go (hunk 2 above)
Comment (abhimutant) [reply nutanix/data_source_nutanix_foundation_central_imaged_clusters_list.go:467]: These are optional fields only.

**Pull request merged**